gtsfm/loader/mobilebrick_loader.py

```diff
@@ -0,0 +1,135 @@
+"""Simple loader class that reads from the MobileBrick dataset.
+
+Reference to MobileBrick: https://code.active.vision/MobileBrick/, Kejie Li et al.
+
+Authors: Akshay Krishnan
+"""
+
+import os
+from pathlib import Path
+from typing import List, Optional
+
+import numpy as np
+
+from gtsam import Cal3Bundler, Pose3, Rot3
+
+import gtsfm.utils.io as io_utils
+import gtsfm.utils.logger as logger_utils
+from gtsfm.common.image import Image
+from gtsfm.loader.loader_base import LoaderBase
+
+logger = logger_utils.get_logger()
+
+
+class MobilebrickLoader(LoaderBase):
+    """Loader class that reads from the MobileBrick dataset."""
+
+    def __init__(
+        self,
+        data_dir: str,
+        max_frame_lookahead: int = 5,
+        max_resolution: int = 1024,
+        input_worker: Optional[str] = None,
+    ) -> None:
+        """ """
+        super().__init__(max_resolution=max_resolution, input_worker=input_worker)
+
+        self._max_frame_lookahead = max_frame_lookahead
+        self._image_dir = os.path.join(data_dir, "image")
+        self._num_images = len(os.listdir(self._image_dir))
+
+        # Cache image paths
+        self._image_paths = []
+        for i in range(self._num_images):
+            image_path = os.path.join(self._image_dir, f"{i:06d}.jpg")
+            self._image_paths.append(image_path)
+
+        # Load GT intrinsics
+        self._intrinsics_dir = os.path.join(data_dir, "intrinsic")
+        self._intrinsics = []
+        for i in range(self._num_images):
+            intrinsics_file = os.path.join(self._intrinsics_dir, f"{i:06d}.txt")
+            K = np.loadtxt(intrinsics_file)
+            self._intrinsics.append(Cal3Bundler((K[0, 0] + K[1, 1]) / 2, 0, 0, K[0, 2], K[1, 2]))
```

Comment:
nit: keyword args on the Cal3Bundler args could make this a little more readable to me, but up to you.